### PR TITLE
Dev oleguer

### PR DESCRIPTION
**Pull request by OleguerCanal**
Testing a bit scala

### CODE REVIEW

File: src/main/scala/Comparator.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  /** Get Jaccard similarity between two sorted sets
+   * @param s1 First sorted set
+   * @param s2 Second sorted set
+   * @return Jaccard similarity (s1, s2)
+   */
+  def getJaccardSim(s1 : scala.collection.SortedSet[Int], s2 : scala.collection.SortedSet[Int]) : Float = {
+    val intersection = s1.intersect(s2)
+    val union = s1.union(s2)
+    return intersection.size.toFloat/union.size.toFloat
```
Comment (pmbrull) [inline src/main/scala/Comparator.scala]: Return statements in Scala are not idiomatic!

File: src/main/scala/Main.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import org.apache.spark.sql.SparkSession
+import org.apache.log4j.Logger
+import org.apache.log4j.Level
+
+import shingler.Shingler
+import hashing.Hasher
+import hashing.MinHasher
+import comparator.Comparator
+
+object Main {
```
Comment (pmbrull) [inline src/main/scala/Main.scala:12]: Scala has an App extension that can be added to an object and it takes care of the "def main..." in a cleaner way

File: src/main/scala/Shingler.scala [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  * @param shingle_len the length (e.g. in characters) in which to split the document
+  * @param hash_bins the number of bins into which to map the shingle hashes
+  */
+class Shingler(shingle_len: Int, hash_bins : Int) {
+
+  /** Split document into shingles
+    * @param doc Document which needs to be split
+    * @return Set of shingles found in document
+    */
+  def getShingles(doc : String) : Set[String] = {
```
Comment (pmbrull) [inline src/main/scala/Shingler.scala]: In general, functions could be rewritten as "def getShingles(doc: String): Set[String] = blabla. One-liners do not need braces, and ":" should stick to the left.

**Review by pmbrull — COMMENTED** [review]
Hola guerrito, et deixo 4 comentaris general pour le fun, però tela bona feina per ser el primer cop que ho enganxes!!

File: src/main/scala/Comparator.scala (hunk 1 above)
Comment (OleguerCanal) [reply src/main/scala/Comparator.scala]: Meaning I shouldn't write return?

File: src/main/scala/Shingler.scala (hunk 3 above)
Comment (OleguerCanal) [reply src/main/scala/Shingler.scala]: Oh nice!

File: src/main/scala/Comparator.scala (hunk 1 above)
Comment (pmbrull) [reply src/main/scala/Comparator.scala]: exactly. Last element in each function will be returned

**Pull request merged**